cli/deno_dir.rs

```diff
@@ -110,42 +122,17 @@ impl DenoDir {
   // https://github.com/denoland/deno/blob/golang/deno_dir.go#L32-L35
   pub fn cache_path(
     self: &Self,
-    filename: &str,
+    filepath: &Path,
     source_code: &[u8],
   ) -> (PathBuf, PathBuf) {
     let cache_key =
-      source_code_hash(filename, source_code, version::DENO, &self.config);
+      source_code_hash(filepath, source_code, version::DENO, &self.config);
     (
       self.gen.join(cache_key.to_string() + ".js"),
       self.gen.join(cache_key.to_string() + ".js.map"),
     )
   }
 
-  pub fn code_cache(
-    self: &Self,
-    module_meta_data: &ModuleMetaData,
-  ) -> std::io::Result<()> {
-    let (cache_path, source_map_path) = self
-      .cache_path(&module_meta_data.filename, &module_meta_data.source_code);
-    // TODO(ry) This is a race condition w.r.t to exists() -- probably should
-    // create the file in exclusive mode. A worry is what might happen is there
-    // are two processes and one reads the cache file while the other is in the
-    // midst of writing it.
-    if cache_path.exists() && source_map_path.exists() {
-      Ok(())
-    } else {
-      match &module_meta_data.maybe_output_code {
-        Some(output_code) => fs::write(cache_path, output_code),
-        _ => Ok(()),
-      }?;
-      match &module_meta_data.maybe_source_map {
-        Some(source_map) => fs::write(source_map_path, source_map),
-        _ => Ok(()),
-      }?;
-      Ok(())
-    }
-  }
-
   pub fn fetch_module_meta_data_async(
```

Comment:
Looking at current usage of `fetch_module_meta_data_async` (and `fetch_module_meta_data`) I'm considering dropping `referrer` argument - besides `op_fetch_module_meta_data` it is always `"."` and the only place it's useful is in the error message:
```
Cannot resolve module "file:///some/file.ts" from "."
Cannot resolve module "file:///some/other/file.ts" from "file:///some/file.ts"
// would become
Cannot resolve module "file:///some/file.ts"
Cannot resolve module "file:///some/other/file.ts"
```

Since specifier should already be resolved I believe using referrer is redundant - you request local or remote file via absolute path and cache acts like a black box giving it back. This change would bring us closer to refactoring `DenoDir` into actual cache.

CC @ry

Reply:
SGTM. I was thinking this also.